### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by thomwolf**
Fixing `run_squad.py` pre-processing bug.

Various clean-ups:

- the weight initialization was not optimal (tf. truncated_normal_initializer(stddev=0.02) was translated in weight.data.normal_(0.02) instead of weight.data.normal_(mean=0.0, std=0.02) which likely affected the performance of run_classifer.py also.
- gradient accumulation loss was not averaged over the accumulation steps which would have required to change the hyper-parameters for using accumulation.
- the evaluation was not done with torch.no_grad() and thus sub-optimal in terms of speed/memory.

### CODE REVIEW

**Comment by evalstate** [comment]
Trace for this mergeability run: https://huggingface.co/datasets/evalstate/transformers-merge-experiments/blob/main/2604231259-ukQdl3__dev__codex.jsonl
